public/app/plugins/datasource/grafana-azure-monitor-datasource/components/MetricsQueryEditor/NewDimensionFields.tsx

```diff
@@ -0,0 +1,207 @@
+import React, { useEffect, useMemo, useState } from 'react';
+
+import { SelectableValue, DataFrame, PanelData } from '@grafana/data';
+import { AccessoryButton, EditorList } from '@grafana/experimental';
+import { Select, HorizontalGroup, MultiSelect, Field } from '@grafana/ui';
+
+import { AzureMetricDimension, AzureMonitorOption, AzureMonitorQuery, AzureQueryEditorFieldProps } from '../../types';
+
+import { setDimensionFilters } from './setQueryValue';
+
+interface DimensionFieldsProps extends AzureQueryEditorFieldProps {
+  dimensionOptions: AzureMonitorOption[];
+}
+
+interface DimensionLabels {
+  [key: string]: Set<string>;
+}
+
+const useDimensionLabels = (data: PanelData | undefined, query: AzureMonitorQuery) => {
+  const [dimensionLabels, setDimensionLabels] = useState<DimensionLabels>({});
+  useEffect(() => {
+    let labelsObj: DimensionLabels = {};
+    if (data?.series?.length) {
+      // Identify which series' in the dataframe are relevant to the current query
+      const series: DataFrame[] = data.series.flat().filter((series) => series.refId === query.refId);
+      const fields = series.flatMap((series) => series.fields);
+      // Retrieve labels for series fields
+      const labels = fields
+        .map((fields) => fields.labels)
+        .flat()
+        .filter((item) => item!);
+      for (const label of labels) {
+        // Labels only exist for series that have a dimension selected
+        if (label) {
```

Comment:
it's kind of weird that you have to check that the label exists right because didn't we already filter out the false-y ones on line 31?